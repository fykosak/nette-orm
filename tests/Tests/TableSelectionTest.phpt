<?php

declare(strict_types=1);

namespace Fykosak\NetteORM\Tests\Tests;

use Fykosak\NetteORM\Tests\ORM\EventModel;
use Fykosak\NetteORM\Tests\ORM\ParticipantModel;
use Fykosak\NetteORM\Tests\ORM\EventService;
use Fykosak\NetteORM\Tests\ORM\ParticipantService;
use Fykosak\NetteORM\TypedSelection;
use Nette\Database\Table\ActiveRow;
use Nette\Database\Table\Selection;
use Tester\Assert;

require_once __DIR__ . '/TestCase.php';

class TableSelectionTest extends TestCase
{

    public function testType(): void
    {
        /** @var EventService $serviceEvent */
        $serviceEvent = $this->container->getByType(EventService::class);
        $selection = $serviceEvent->getTable();
        Assert::type(TypedSelection::class, $selection);
        Assert::type(Selection::class, $selection);
        $event = $selection->fetch();
        Assert::type(ActiveRow::class, $event);
        Assert::type(EventModel::class, $event);
    }

    public function testReferenced(): void
    {
        /** @var ParticipantService $serviceEvent */
        $serviceEvent = $this->container->getByType(ParticipantService::class);
        $participant = $serviceEvent->getTable()->fetch();

        Assert::type(ParticipantModel::class, $participant);
        $event = $participant->getEvent();
        Assert::type(EventModel::class, $event);
    }

    public function testRelated(): void
    {
        /** @var ParticipantService $serviceEvent */
        $serviceEvent = $this->container->getByType(EventService::class);
        $event = $serviceEvent->getTable()->fetch();
        $row = $event->related('participant')->fetch();

        Assert::false($row instanceof ParticipantModel);
        $participant = ParticipantModel::createFromActiveRow($row);
        Assert::type(ParticipantModel::class, $participant);
    }

    public function testPassModel(): void
    {
        /** @var ParticipantService $serviceEvent */
        $serviceEvent = $this->container->getByType(EventService::class);
        $event = $serviceEvent->getTable()->fetch();
        $newEvent = EventModel::createFromActiveRow($event);
        Assert::same($event, $newEvent);
    }
}

$testCase = new TableSelectionTest();
$testCase->run();
