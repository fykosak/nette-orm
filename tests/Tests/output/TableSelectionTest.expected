Fykosak\NetteORM\Tests\ORM\ModelEvent::__set_state(/* #f2b2 */ [
	'table' => Fykosak\NetteORM\TypedTableSelection::__set_state(/* #25b4 */ [
		'modelClassName' => 'Fykosak\\NetteORM\\Tests\\ORM\\ModelEvent',
		'explorer' => Nette\Database\Explorer::__set_state(/* #ff7d */ [
			'connection' => Nette\Database\Connection::__set_state(/* #497d */ [
				'onConnect' => [],
				'onQuery' => [],
				'params' => ['mysql:host=127.0.0.1;dbname=nette_orm_test', 'root', 'fykos1'],
				'options' => [],
				'driver' => Nette\Database\Drivers\MySqlDriver::__set_state(/* #c223 */ [
					'connection' => /* Nette\Database\Connection dumped on line 5 */,
					'version' => '5.7.33-0ubuntu0.18.04.1',
				]),
				'preprocessor' => Nette\Database\SqlPreprocessor::__set_state(/* #e855 */ [
					'connection' => /* Nette\Database\Connection dumped on line 5 */,
					'driver' => /* Nette\Database\Drivers\MySqlDriver dumped on line 10 */,
					'params' => [
						'SELECT * FROM `participant` WHERE (`participant`.`event_id` IN (?))',
						[1, 2, 3],
					],
					'remaining' => [1, 2, 3],
					'counter' => 2,
					'useParams' => true,
					'arrayMode' => null,
				]),
				'pdo' => PDO::__set_state(/* #3880 */ []),
				'sql' => 'SELECT * FROM `event`',
			]),
			'structure' => Nette\Database\Structure::__set_state(/* #7243 */ [
				'connection' => /* Nette\Database\Connection dumped on line 5 */,
				'cache' => Nette\Caching\Cache::__set_state(/* #6735 */ [
					'storage' => Nette\Caching\Storages\FileStorage::__set_state(/* #fd3c */ [
						'dir' => '../temp',
						'journal' => null,
						'locks' => null,
					]),
					'namespace' => "Nette.Database.Structure.4013f88097dca8446f3aa471806f6968\x00",
				]),
				'structure' => [
					'tables' => [
						['name' => 'event', 'view' => false],
						['name' => 'participant', 'view' => false],
					],
					'columns' => [
						'event' => [
							[
								'name' => 'event_id',
								'table' => 'event',
								'nativetype' => 'INT',
								'size' => 11,
								'nullable' => false,
								'default' => null,
								'autoincrement' => true,
								'primary' => true,
								'vendor' => [
									'field' => 'event_id',
									'type' => 'int(11)',
									'collation' => null,
									'null' => 'NO',
									'key' => 'PRI',
									'default' => null,
									'extra' => 'auto_increment',
									'privileges' => 'select,insert,update,references',
									'comment' => '',
								],
							],
							[
								'name' => 'begin',
								'table' => 'event',
								'nativetype' => 'DATETIME',
								'size' => null,
								'nullable' => false,
								'default' => null,
								'autoincrement' => false,
								'primary' => false,
								'vendor' => [
									'field' => 'begin',
									'type' => 'datetime',
									'collation' => null,
									'null' => 'NO',
									'key' => '',
									'default' => null,
									'extra' => '',
									'privileges' => 'select,insert,update,references',
									'comment' => '',
								],
							],
							[
								'name' => 'end',
								'table' => 'event',
								'nativetype' => 'DATETIME',
								'size' => null,
								'nullable' => false,
								'default' => null,
								'autoincrement' => false,
								'primary' => false,
								'vendor' => [
									'field' => 'end',
									'type' => 'datetime',
									'collation' => null,
									'null' => 'NO',
									'key' => '',
									'default' => null,
									'extra' => '',
									'privileges' => 'select,insert,update,references',
									'comment' => '',
								],
							],
						],
						'participant' => [
							[
								'name' => 'participant_id',
								'table' => 'participant',
								'nativetype' => 'INT',
								'size' => 11,
								'nullable' => false,
								'default' => null,
								'autoincrement' => true,
								'primary' => true,
								'vendor' => [
									'field' => 'participant_id',
									'type' => 'int(11)',
									'collation' => null,
									'null' => 'NO',
									'key' => 'PRI',
									'default' => null,
									'extra' => 'auto_increment',
									'privileges' => 'select,insert,update,references',
									'comment' => '',
								],
							],
							[
								'name' => 'event_id',
								'table' => 'participant',
								'nativetype' => 'INT',
								'size' => 11,
								'nullable' => false,
								'default' => null,
								'autoincrement' => false,
								'primary' => false,
								'vendor' => [
									'field' => 'event_id',
									'type' => 'int(11)',
									'collation' => null,
									'null' => 'NO',
									'key' => 'MUL',
									'default' => null,
									'extra' => '',
									'privileges' => 'select,insert,update,references',
									'comment' => '',
								],
							],
							[
								'name' => 'name',
								'table' => 'participant',
								'nativetype' => 'VARCHAR',
								'size' => 64,
								'nullable' => true,
								'default' => null,
								'autoincrement' => false,
								'primary' => false,
								'vendor' => [
									'field' => 'name',
									'type' => 'varchar(64)',
									'collation' => 'latin1_swedish_ci',
									'null' => 'YES',
									'key' => '',
									'default' => null,
									'extra' => '',
									'privileges' => 'select,insert,update,references',
									'comment' => '',
								],
							],
						],
					],
					'primary' => ['event' => 'event_id', 'participant' => 'participant_id'],
					'belongsTo' => ['participant' => ['event_id' => 'event']],
					'hasMany' => ['event' => ['participant' => ['event_id']]],
				],
				'isRebuilt' => false,
			]),
			'conventions' => Nette\Database\Conventions\DiscoveredConventions::__set_state(/* #45ff */ [
				'structure' => /* Nette\Database\Structure dumped on line 29 */,
			]),
			'cacheStorage' => /* Nette\Caching\Storages\FileStorage dumped on line 32 */,
		]),
		'context' => /* Nette\Database\Explorer dumped on line 4 */,
		'conventions' => /* Nette\Database\Conventions\DiscoveredConventions dumped on line 182 */,
		'cache' => null,
		'sqlBuilder' => Nette\Database\Table\SqlBuilder::__set_state(/* #1326 */ [
			'tableName' => 'event',
			'conventions' => /* Nette\Database\Conventions\DiscoveredConventions dumped on line 182 */,
			'delimitedTable' => '`event`',
			'select' => [],
			'where' => [],
			'joinCondition' => [],
			'conditions' => [],
			'parameters' => [
				'select' => [],
				'joinCondition' => [],
				'where' => [],
				'group' => [],
				'having' => [],
				'order' => [],
				'joinConditionSorted' => [],
			],
			'order' => [],
			'limit' => null,
			'offset' => null,
			'group' => '',
			'having' => '',
			'reservedTableNames' => ['event' => 'event'],
			'aliases' => [],
			'currentAlias' => null,
			'driver' => /* Nette\Database\Drivers\MySqlDriver dumped on line 10 */,
			'structure' => /* Nette\Database\Structure dumped on line 29 */,
			'cacheTableList' => null,
			'expandingJoins' => [],
		]),
		'name' => 'event',
		'primary' => 'event_id',
		'primarySequence' => false,
		'rows' => [
			1 => /* Fykosak\NetteORM\Tests\ORM\ModelEvent dumped on line 1 */,
			Fykosak\NetteORM\Tests\ORM\ModelEvent::__set_state(/* #431e */ [
				'table' => /* Fykosak\NetteORM\TypedTableSelection dumped on line 2 */,
				'data' => [
					'event_id' => 2,
					'begin' => Nette\Utils\DateTime::__set_state(/* #ab17 */ [
						'date' => '2010-02-01 00:00:00.000000',
						'timezone_type' => 3,
						'timezone' => 'Europe/Prague',
					]),
					'end' => Nette\Utils\DateTime::__set_state(/* #fccd */ [
						'date' => '2010-03-01 00:00:00.000000',
						'timezone_type' => 3,
						'timezone' => 'Europe/Prague',
					]),
				],
				'dataRefreshed' => false,
			]),
			Fykosak\NetteORM\Tests\ORM\ModelEvent::__set_state(/* #672e */ [
				'table' => /* Fykosak\NetteORM\TypedTableSelection dumped on line 2 */,
				'data' => [
					'event_id' => 3,
					'begin' => Nette\Utils\DateTime::__set_state(/* #65ec */ [
						'date' => '2010-03-01 00:00:00.000000',
						'timezone_type' => 3,
						'timezone' => 'Europe/Prague',
					]),
					'end' => Nette\Utils\DateTime::__set_state(/* #9234 */ [
						'date' => '2010-04-01 00:00:00.000000',
						'timezone_type' => 3,
						'timezone' => 'Europe/Prague',
					]),
				],
				'dataRefreshed' => false,
			]),
		],
		'data' => [
			1 => /* Fykosak\NetteORM\Tests\ORM\ModelEvent dumped on line 1 */,
			/* Fykosak\NetteORM\Tests\ORM\ModelEvent dumped on line 225 */,
			/* Fykosak\NetteORM\Tests\ORM\ModelEvent dumped on line 242 */,
		],
		'dataRefreshed' => false,
		'globalRefCache' => ['' => null],
		'refCache' => null,
		'generalCacheKey' => null,
		'specificCacheKey' => null,
		'aggregation' => [],
		'accessedColumns' => ['event_id' => true],
		'previousAccessedColumns' => null,
		'observeCache' => /* Fykosak\NetteORM\TypedTableSelection dumped on line 2 */,
		'keys' => [],
	]),
	'data' => [
		'event_id' => 1,
		'begin' => Nette\Utils\DateTime::__set_state(/* #f848 */ [
			'date' => '2010-01-01 00:00:00.000000',
			'timezone_type' => 3,
			'timezone' => 'Europe/Prague',
		]),
		'end' => Nette\Utils\DateTime::__set_state(/* #c5b9 */ [
			'date' => '2010-02-01 00:00:00.000000',
			'timezone_type' => 3,
			'timezone' => 'Europe/Prague',
		]),
	],
	'dataRefreshed' => false,
])
